be/src/storage/push_handler.cpp

```diff
@@ -457,7 +457,7 @@ Status PushHandler::_delete_convert(const TabletSharedPtr& cur_tablet, RowsetSha
         DCHECK(!_request.__isset.http_file_path);
 
         // 2. init RowsetBuilder of cur_tablet for current push
-        RowsetWriterContext context(kDataFormatUnknown, config::storage_format_version);
```

Comment:
If the config::storage_format_version == 1, you should also write using the v2 format.